elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs

```diff
@@ -93,22 +134,22 @@ callisto targetCount updateEveryNCycles dataCounts =
   -- https://github.com/bittide/Callisto.jl/blob/e47139fca128995e2e64b2be935ad588f6d4f9fb/demo/pulsecontrol.jl#L24
   --
   -- the constants here are chosen to match the above code.
-  k_p, k_i, fStep :: forall m. DSignal dom m Float
+  k_p, k_i, fStep :: forall d. DSignal dom d Float
   k_p = pure 2e-4
   k_i = pure 1e-11
   fStep = pure 5e-4
 
   r_k :: DSignal dom F.FromS32DefDelay Float
   r_k = F.fromS32 $ D.fromSignal $
     let
-      -- These bit coerces are fairly dangerous / unchecked.
-      --
-      -- TODO: Defend datatypes (why they don't overflow), or propagate errors up.
-      measuredSum = bitCoerce . sum <$> dataCounts
-      targetCountS = bitCoerce targetCount
       nBuffers = natToNum @n
+      measuredSum = sumTo32 <$> dataCounts
+      targetCountSigned =
+        case euclid3 @n @m @32 of
+          Dict ->
```

Comment:
Is this not where you'd want to use `safesum`?

Reply:
Yeah, `sumTo32` uses `safeSum`. I figured that was clearer than to chain a bunch of operations right there.

Reply:
Or do you mean for `targetCountSigned`?